algoliasearch/src/main/java/com/algolia/search/saas/BaseIndex.java

```diff
@@ -330,24 +332,37 @@ protected void deleteByQuery(Query query) throws AlgoliaException {
 
     /**
      * Search inside the index
+     *
      * @return a JSONObject containing search results
      * @throws AlgoliaException
      */
-    protected JSONObject search(Query query) throws AlgoliaException {
-        String paramsString = query.getQueryString();
-        if (paramsString.length() > 0) {
-            return client.getRequest("/1/indexes/" + encodedIndexName + "?" + paramsString, true);
-        } else {
-            return client.getRequest("/1/indexes/" + encodedIndexName, true);
+    protected JSONObject search(@NonNull Query query) throws AlgoliaException {
+        String cacheKey = null;
+        String jsonStr = null;
+        if (isCacheEnabled) {
+            cacheKey = String.format("%s", query.getQueryString());
+            jsonStr = searchCache.get(cacheKey);
+        }
+        try {
+            if (jsonStr == null) {
+                jsonStr = new String(searchRaw(query), "UTF-8");
+                if (isCacheEnabled) {
+                    searchCache.put(cacheKey, jsonStr);
```

Comment:
I would rather store the response in the cache when we are sure it parses correctly.